symfony/plugins/orangehrmCorePlugin/Traits/ValidatorTrait.php

```diff
@@ -0,0 +1,40 @@
+<?php
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Core\Traits;
+
+use OrangeHRM\Core\Api\V2\Exception\InvalidParamException;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Api\V2\Validator\Validator;
+use OrangeHRM\Core\Api\V2\Validator\ValidatorException;
+
+trait ValidatorTrait
+{
+    /**
+     * @param array $values
+     * @param ParamRuleCollection|null $rules
+     * @return bool
+     * @throws InvalidParamException
+     * @throws ValidatorException
+     */
+    protected function validate(array $values, ?ParamRuleCollection $rules = null): bool
+    {
+        return Validator::validate($values, $rules);
```

Comment:
What's the upside of trait calling Validator::validate ?

Reply:
to reduce static reference. since static usage hard to mock, but this Validator not using frequently here and there.